### PR TITLE
feat(wizard-dialog): wizard-dialog content definition through `WizardInput` objects

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Hi all, finally the PR that will extend the `Wizard` API to allow content definition through `WizardInput` objects definition instead of `TemplateResult`. 

The advantage is that a plug-in author in JavaScript can trigger a wizard-dialog in without loading lit-element. 
The disadvantage is that with `WizardInput` you can only define `Select`, `TextField` and `Checkbox` kind inputs. With `TemplateResult` we can add list, buttons .... basically all kind of web-components. This is why we added the WizardInput content definition on top of the old API.

### CODE REVIEW

File: test/unit/wizard-dialog.test.ts [hunk 1]
```diff
@@ -258,4 +262,371 @@ describe('wizard-dialog', () => {
...
+
+      it('makes sure to not render nullSwitch with missing/false nullable', () =>
+        expect(select.nullSwitch).to.not.exist);
+
+      it('makes sure that wizard-select has correct label', () =>
+        expect(select.label).to.be.equal('myLabel'));
+
+      it('makes sure that wizard-select has correct maybeValue', async () =>
+        expect(select.maybeValue).to.equal('multi1'));
+    });
```
Comment (dlabordus) [inline test/unit/wizard-dialog.test.ts:389]: Should there be a test to see that there are 2 options available for the "values" passed. (There are 2 'mwc-list-item' under 'wizard-select')

File: src/foundation.ts [hunk 2]
```diff
@@ -171,7 +171,7 @@ export function newActionEvent<T extends EditorAction>(
 
 export const wizardInputSelector =
   'wizard-textfield, mwc-textfield, ace-editor, mwc-select,wizard-select, wizard-checkbox';
-export type WizardInput =
+export type WizardInputElement =
```
Comment (dlabordus) [inline src/foundation.ts:174]: Isn't the WizardCheckbox missing here?

**Review by dlabordus — CHANGES_REQUESTED** [review]

File: test/unit/wizard-dialog.test.ts (hunk 1 above)
Comment (JakobVogelsang) [reply test/unit/wizard-dialog.test.ts:389]: I think this is covered with the snapshot test above. Here the list-items reflect the selection.

File: src/foundation.ts (hunk 2 above)
Comment (JakobVogelsang) [reply src/foundation.ts:174]: True it is missing I will create an extra PR that points to this on. For now I think for others easier to review, if not mixed with the changes here :) We can keep it open until the PR is merged.

**Review by dlabordus — APPROVED** [review]

File: test/unit/wizard-dialog.test.ts [hunk 3]
```diff
@@ -258,4 +262,371 @@ describe('wizard-dialog', () => {
       after(() => localStorage.removeItem('mode'));
     });
   });
+
+  describe('with content definition throught WizardInput s', () => {
```
Comment (danyill) [inline test/unit/wizard-dialog.test.ts:266]:
```suggestion
  describe('with content definition throught WizardInputs', () => {
```

I'm better with spelling than code.

File: test/unit/wizard-dialog.test.ts [hunk 4]
```diff
@@ -258,4 +262,371 @@ describe('wizard-dialog', () => {
...
+    describe('for a specific WizardInputs of the kind Checkbox', () => {
```
Comment (danyill) [inline test/unit/wizard-dialog.test.ts:267]:
```suggestion
    describe('for a specific WizardInput of the kind Checkbox', () => {
```

**Review by danyill — COMMENTED** [review]
Christopher Robbin once patted my head and said "You are only a :bear: with a very small :brain:, so let's imagine, for now, that the review process is more for your education than feedback".

On such a basis I can say I like this idea.

**Pull request merged**